### ISSUE TITLE
Array of items is duplicated on model after update

### ISSUE THREAD

**Issue by jaivia** [aws-amplify/amplify-ios#545]
**Describe the bug**
When model as an Array of items (in test case type was [String]?) updating using Amplify.API.mutate(.update(model)) the new model appends all contents of the array to the existing model creating duplicates. Updating a model will cause exponential growth of all array properties.

**To Reproduce**
Create a schema with [String]? property.
Create a new model with this property populated and sync
Update this model, with or without changing any value

- The value of the array property on the model returned will have duplicated values.

**Expected behavior**
Models returned by update query should match what was updated

**Environment(please complete the following information):**
 - Amplify Framework Version: 4.21.1
 - Dependency Manager: cocoapods
 - Swift Version : 5.0

**Device Information (please complete the following information):**
 - Device: Simulator
 - iOS Version:  iOS 11
 - Specific to simulators:

**Comment by lawmicha** [comment]
Hi @jaivia, thanks for reporting this issue. I tried to reproduce this but could not. The schema i'm using is
```
type Todo @model {
  id: ID!
  colors: [String]
}
```
I tried creating a todo, then querying for the todo by Id, and updating the queried data. the todo's colors is not duplicated. I also tried creating a todo, and directly using the locally created todo and updating it.

Are you able to reproduce this in Amplify 1.0.4 and Amplify CLI 4.22.0? There were a few changes related schemas containing an array of types such as this one. So once you upgrade to 4.22.0, you will have to re-run `amplify codegen models` to regenerate the new models. Make sure you are on the latest Amplify Library by running `pod cache clean --all && pod install --repo-update` and checking your Podfile.lock is using Amplify 1.0.4 libraries.
